From 410a062e9e2733df07ccc060418b21dc50bec8ad Mon Sep 17 00:00:00 2001
From: donghuawei <donghuawei@eswincomputing.com>
Date: Tue, 24 Dec 2024 16:02:46 +0800
Subject: [PATCH 354/652] WIN2030-16857:fix:dsp pm devfreq bug fix

Changelogs:
	dsp pm devfreq bug fix when not set CONFIG_PM_DEVFREQ

Change-Id: I20778079fa9fd29ff08803284c59441dc443e84b
Signed-off-by: donghuawei <donghuawei@eswincomputing.com>
---
 drivers/soc/eswin/ai_driver/dsp/dsp_main.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/soc/eswin/ai_driver/dsp/dsp_main.c b/drivers/soc/eswin/ai_driver/dsp/dsp_main.c
index 1a5a1a0d7e16e..5a2f3e97d9d8b 100644
--- a/drivers/soc/eswin/ai_driver/dsp/dsp_main.c
+++ b/drivers/soc/eswin/ai_driver/dsp/dsp_main.c
@@ -884,6 +884,7 @@ static void dsp_init_prio_array(struct es_dsp *dsp)
 	set_bit(DSP_MAX_PRIO, array->bitmap);
 }
 
+#if defined(CONFIG_PM_DEVFREQ)
 /* devfreq target function to set frequency */
 static int dsp_devfreq_target(struct device *dev, unsigned long *freq, u32 flags)
 {
@@ -914,6 +915,7 @@ static struct devfreq_dev_profile dsp_devfreq_profile = {
 	.target = dsp_devfreq_target,
 	.get_cur_freq = dsp_devfreq_get_cur_freq,
 };
+#endif
 
 static int32_t  dsp_probe_result = 0;
 static int es_dsp_hw_probe(struct platform_device *pdev)
@@ -921,7 +923,9 @@ static int es_dsp_hw_probe(struct platform_device *pdev)
 	int ret;
 	char nodename[sizeof("es-dsp") + 3 * sizeof(int)];
 	struct es_dsp *dsp;
+#if defined(CONFIG_PM_DEVFREQ)
 	struct devfreq *df;
+#endif
 
 	dsp = devm_kzalloc(&pdev->dev,
 			   sizeof(*dsp) + sizeof(struct es_dsp_stats) +
@@ -989,6 +993,7 @@ static int es_dsp_hw_probe(struct platform_device *pdev)
 		goto err_mbox_clk;
 	}
 
+#if defined(CONFIG_PM_DEVFREQ)
 	/* Add OPP table from device tree */
 	ret = dev_pm_opp_of_add_table(&pdev->dev);
 	if (ret) {
@@ -999,8 +1004,10 @@ static int es_dsp_hw_probe(struct platform_device *pdev)
 	df = devm_devfreq_add_device(&pdev->dev, &dsp_devfreq_profile, "userspace", NULL);
 	if (IS_ERR(df)) {
 		dsp_err("%s, %d, add devfreq failed\n", __func__, __LINE__);
+		ret = PTR_ERR(df);
 		goto err_dsp_devfreq;
 	}
+#endif
 
 	ret = es_dsp_clk_enable(dsp);
 	if (ret) {
@@ -1063,8 +1070,10 @@ static int es_dsp_hw_probe(struct platform_device *pdev)
 err_tbu_power:
 	es_dsp_clk_disable(dsp);
 err_dsp_clk:
+#if defined(CONFIG_PM_DEVFREQ)
 	devm_devfreq_remove_device(dsp->dev, df);
 err_dsp_devfreq:
+#endif
 	dsp_disable_mbox_clock(dsp);
 err_mbox_clk:
 	es_dsp_unmap_resource(dsp);
-- 
2.51.0

